### PR TITLE
Allow type aliases to omit 'declare' keyword in '.d.ts' files

### LINKED ISSUE #4487: 'declare' modifier required on top-level type alias in a .d.ts file
```TypeScript
type Foo = number;
```

Save this as a .d.ts file, you'll get an error about how a `declare` modifier is required in a top-level declaration.

### PR DESCRIPTION
**Pull request by maybejulian**
Fixes #4487

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -15917,6 +15917,7 @@ namespace ts {
             //     export_opt   AmbientDeclaration
             //
             if (node.kind === SyntaxKind.InterfaceDeclaration ||
+                node.kind === SyntaxKind.TypeAliasDeclaration ||
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: Even though it's not part of the spec, can you amend the grammar in the comment to reflect the change, and then add a `// TODO: The spec needs to be amended to reflect this grammar.` below the grammar?

**Comment by mhegazy** [comment]
:+1:

**Comment by mhegazy** [comment]
thanks @maybejulian!

**Comment by DanielRosenwasser** [comment]
Julian's first PR :star2: :star2: :star2: :star2: :star2:

**Comment by maybejulian** [comment]
Time to celebrate :balloon: :beers:

**Pull request merged**